app/controllers/users_controller.rb

```diff
@@ -11,27 +11,17 @@ def index
     redirect_to search_path(type: 'users')
   end
 
-  def verify
-    if current_user.present? && current_user == User.where(id: params[:id]).first && params[:token] == current_user.perishable_token
-      current_user.update_column(:verified_email, true)
-      flash[:notice] = 'Thank you. Your account has been verified. You may now contribute to H2O.'
-      redirect_to user_path(current_user)
-    elsif current_user.present?
-      flash[:notice] = 'Your account has not been verified. Please try again by requesting an email verification <a href="' + verify_user_url(current_user)  + '" target="blank">here</a>.'
-      redirect_to user_path(current_user)
-    else
-      flash[:notice] = 'Your account has not been verified. Please login and try visiting the link in the email again.'
-      redirect_to '/user_sessions/new'
-    end
-  end
-
   def create
-    @user = User.new(permitted_user_params)
+    user_params = permitted_user_params
+    temporary_password = Random.new_seed
```

Comment:
Authlogic has the variable `require_password_confirmation`. 

To save a user without having a password it'd have to be set to false. But then, when the user does add a password there is no such thing as the password_confirmation field. 

So I added a temp password to the newly created user account so that we still have the password + password confirmation fields